ui/components/src/StatChart/calculateFontSize.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+
+interface CalculateFontSize {
+  text: string;
+  fontWeight: number;
+  width: number;
+  height: number;
+  lineHeight?: number; // default is 1.2
+  maxSize?: number;
+}
+
+const DEFAULT_LINE_HEIGHT = 1.2;
+
+let canvasContext: CanvasRenderingContext2D | null;
+
+function useCanvasContext() {
+  if (!canvasContext) {
+    canvasContext = document.createElement('canvas').getContext('2d');
```

Comment:
Do you happen to know if there's any perf issues with creating these over and over and not cleaning them up? I _think_ JS is smart about garbage collecting them if they're never inserted into the dom, but it'd be worth double-checking.

Reply:
If we are doing something global, I would make it pretty clear with the name + a comment. The naming convention seems like a hook with the `use` prefix, so maybe `getGlobalCanvasContext()` instead since no react hooks are used.